app/src/main/java/protect/card_locker/ManageGroupActivity.java

```diff
@@ -0,0 +1,315 @@
+        }
+        mAdapter = new ManageGroupCursorAdapter(this, null, this, mGroup);
+        mCardList.setAdapter(mAdapter);
+        registerForContextMenu(mCardList);
+
+        if (inputSavedInstanceState != null) {
+            ManageGroupActivityInGroupState adapterState = inputSavedInstanceState.getParcelable("mAdapterState");
+            if (adapterState != null) {
+                mAdapterState = adapterState.getMap();
+            }
+            mCurrentGroupName = inputSavedInstanceState.getString("mCurrentGroupName");
+        }
+
+        mActionBar = getSupportActionBar();
+        if (mActionBar == null){
+            throw(new RuntimeException("mActionBar is not expected to be null here"));
+        }
+        mActionBar.setDisplayHomeAsUpEnabled(true);
+        mActionBar.setDisplayShowHomeEnabled(true);
+
+        mSaveButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String currentGroupName = mGroupNameText.getText().toString();
+                if(!currentGroupName.trim().equals(mGroup._id)){
+                    if(!mGroupNameNotInUse) {
+                        Toast toast = Toast.makeText(getApplicationContext(), R.string.group_name_already_in_use, Toast.LENGTH_SHORT);
+                        toast.show();
+                        return;
+                    }
+                    if(currentGroupName.trim().length() == 0){
+                        Toast toast = Toast.makeText(getApplicationContext(), R.string.group_name_is_empty, Toast.LENGTH_SHORT);
+                        toast.show();
+                        return;
+                    }
+                }
+
+                mAdapter.commitToDatabase(getApplicationContext());
+                Toast toast = Toast.makeText(getApplicationContext(), R.string.group_updated, Toast.LENGTH_SHORT);
+                if(!currentGroupName.trim().equals(mGroup._id)){
+                    mDB.updateGroup(mGroup._id, currentGroupName.trim());
+                }
+                toast.show();
+                finish();
+            }
+        });
+    }
+
+    private void resetGroupNameTextColor() {
+        mGroupNameText.setTextColor(mGroupNameInputTextColor);
+    }
+
+    private void checkIfGroupNameIsInUse(){
+        mGroupNameNotInUse = false;
+        String currentGroupName = mGroupNameText.getText().toString();
+        if (!mGroup._id.equals(currentGroupName.trim())) {
+            Group group = mDB.getGroup(currentGroupName.trim());
+            if (group != null) {
+                mGroupNameNotInUse = false;
+                mGroupNameText.setTextColor(Color.RED);
+            } else {
+                mGroupNameNotInUse = true;
+            }
+        }
+    }
+
+    @Override
+    protected void onResume()
+    {
+        super.onResume();
+
+        setTitle(getString(R.string.edit) + ": " + mGroup._id);
```

Comment:
Should use a formatted string instead, look at https://github.com/TheLastProject/Catima/blob/6ad99d47aa72156860c4def2a329be8c8051b073/app/src/main/res/values/strings.xml#L141 for example.

https://github.com/TheLastProject/Catima/blob/92da7923789d235de4add71d39e8cdcaccc71f1f/app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java#L456